web_src/js/features/repo-editor.js

```diff
@@ -105,8 +105,8 @@ export function initRepoEditor() {
       $section.last().remove();
       $divider.last().remove();
     }
-    if (e.keyCode === 191) {
-      parts = $(this).val().split('/');
+    parts = $(this).val().split('/');
+    if (e.keyCode === 191 || parts.length > 1) {
```

Comment:
Maybe 
```suggestion
    if (parts.length > 1) {
```
is enough?

I'm not sure, haven't tested it.

Reply:
And how about only handle "input" event?  Then `right click -> paste` also works.

Reply:
> And how about only handle "input" event? Then `right click -> paste` also works.

I tried to change `$editFilename.on('keyup',..)` to `$editFilename.on('input',..)` , so right click paste also works, but i am not very sure why another "keyup" event is triggered in the end, not sure if i can remove the triggered "keyup" event. The behaviors look the same if i remove the triggered "keyup"

Reply:
I found the "input" event do not have  e.keyCode so the backspace logic will not work. So may handle both "input" and "keyup" events.

Reply:
Sorry I didn't get the point. The latest code uses 'keyup input' together? It look a little strange, it's better not mixing different events with the same event listener.

I think the `keyup(backspace)` and `input` are for different purposes, and they could be different event listeners:

* `input`:  if the content is non-empty,  try to split the path, then it works for all cases.
* `keyup(backspace)`: if the cursor is at the beginning of the input field, remove the last slash.

Reply:
Having a listener for both events will certainly trigger the handler two times on keyboard input. Better to separate them. There is also a `paste` event which may be useful.

Reply:
> Having a listener for both events will certainly trigger the handler two times on keyboard input. Better to separate them. There is also a `paste` event which may be handy for what this is trying to do.

Tested paste, and it will need extra steps of binding the input and reading them, looks like input is easier to use in this case.

Reply:
Yup, I also think "input" is enough. "paste" is only necessary for non-text content (like file/image), it's not for this case.